### DISCUSSION TITLE
Observation: this integration retrieves Solcast forecast as published by Solcast (was "Forecast apparently does not incorporate local weather")

### DISCUSSION THREAD

**Discussion by mtbsteve** [BJReplay/ha-solcast-solar#205] [General]
I am using Solcast for a while now. What I experienced is that the forecast and real production values reasonably match on sunny days, however deviate by a factor of 2 or more when the sky is overcast or even rainy. Home location is correctlty set, and I applied the correction factors in a way so that the forecast is matching the production on clear skies.
What weather data are you using in Solcast? The weather information in HA coming from the Meteorologisk institutt (Met.no) is perfectly matching my local weather conditions at home.

**Comment by gcoan** [comment]
This is a question for Solcast themselves, not the integration which just pulls the solcast forecast data.

Have a read of their website and it'll tell you how they have a network of weather satellites to predict cloud cover and local weather.

In my experience I pretty much always achieve somewhere around the PV50 forecast each day. Sometimes it is towards the PV10, but usually it's towards the PV50.

At the end of the day its just a forecast

**Comment by BJReplay** [comment]
As @gcoan said, this is a question (if it is a question for anyone) for Solcast.

We (the maintainers of this integration) are not related to Solcast.

You might find that the 10% forecast is just as accurate on a sunny day, (after adjusting the factors as you describe above) and far more accurate on cloudy days.

**Comment by autoSteve** [comment]
If fact, on sunny days, like this chart today for the afternoon only which was without a cloud, there is a remarkable tracking of forecast. Very late afternoon here I need to adjust my dampening because building blocking the sun... (The flat top of generation is my disti limiting export.)

Cloud gets difficult for anyone to forecast expected generation, and Solcast is usually somewhere between the tenth and fiftieth percentile as @gcoan wrote.

My automations base some actions on the question, "Will I likely have a surplus of generation?", which is what I really care about. I do not want to leave generation on the table, given export limits. So am I expecting a surplus above X? Run the pool filter and heater! Put on the washer! Run the clothes dryer! Put on all of the dishwashers! (Wall displays show what is a good thing to do, and automation runs the pool.) Do it at whatever time of the day, because there will be an expected surplus that won't be used to fill the battieries any more than full.

It usually works brilliantly. Today, the partly cloudy and showery morning let down the math. It was well under 50th percentile. But the overall day was true to the forecast word. I could have ran any appliance during the day at any time and would have achieved exactly the same power bill. Zero.

In fact, they paid me.

<img width="474" alt="image" src="https://github.com/user-attachments/assets/4ee3fdd3-ad99-4219-abda-7b54fc8d689b">
